plugin/trace/setup.go

```diff
@@ -35,7 +35,9 @@ func traceParse(c *caddy.Controller) (*trace, error) {
 	)
 
 	cfg := dnsserver.GetConfig(c)
-	tr.serviceEndpoint = cfg.ListenHosts[0] + ":" + cfg.Port
+	if cfg.ListenHosts[0] != "" {
+		tr.serviceEndpoint = cfg.ListenHosts[0] + ":" + cfg.Port
```

Comment:
should we *not* add the port here? As you say in the description this doesn't work with zipkin...

Reply:
Zipkin does understand the port when associated with a resolvable hostname / an IP, and there is always a port in the config.

I assume to track the the correct listener you need the right couple of bind + port

That would also work with no port but would be less precise